unattended.sh

```diff
@@ -1,186 +1,803 @@
-yum -q clean all
-rpm -q centos-release-scl > /dev/null 2>&1
-if [ "x$?" '!=' x0 ] ; then
-  yum -q install -y centos-release-scl
-  if [ "x$?" '!=' x0 ] ; then
-    error_and_exit "Could not install Software Collections repository (package centos-release-scl)"
-  fi
-fi
-rpm -q centreon-release-21.04 > /dev/null 2>&1
-if [ "x$?" '!=' x0 ] ; then
-  yum -q install -y --nogpgcheck http://yum.centreon.com/standard/21.04/el7/stable/noarch/RPMS/centreon-release-21.04-2.el7.centos.noarch.rpm
-  if [ "x$?" '!=' x0 ] ; then
-    error_and_exit "Could not install Centreon repository"
-  fi
-fi
-print_step_end
+	log "INFO" "Check if the system OS is supported and set the environment variables"
 
-# Get the Centos version
-eval $(cat /etc/os-release | grep VERSION_ID)
+	get_os_information
 
-if [ $VERSION_ID == 8 ] ; then
-  # Switch dnf upstream to install php73
-  dnf module enable php:7.3
-fi
+	case "$detected_os_version" in
+	7*)
+		log "INFO" "Setting specific part for v7 ($detected_os_version)"
 
+		case "$detected_os_release" in
+		centos-release* | centos-linux-release*)
+			BASE_PACKAGES=(centos-release-scl)
+			;;
+
+		oraclelinux-release* | enterprise-release*)
+			BASE_PACKAGES=(oraclelinux-release-el7)
+			;;
+		esac
+		RELEASE_RPM_URL="http://yum.centreon.com/standard/$CENTREON_MAJOR_VERSION/el7/stable/noarch/RPMS/centreon-release-$CENTREON_RELEASE_VERSION.el7.centos.noarch.rpm"
+		log "INFO" "Install Centreon from ${RELEASE_RPM_URL}"
+		PHP_BIN="/opt/rh/rh-php73/root/bin/php"
+		PHP_ETC="/etc/opt/rh/rh-php73/php.d/"
+		OS_SPEC_SERVICES="rh-php73-php-fpm httpd24-httpd"
+		PKG_MGR="yum"
+
+		set_centreon_repos
+
+		log "INFO" "Installing required base packages"
+		if ! $PKG_MGR -y -q install ${BASE_PACKAGES[@]}; then
+			error_and_exit "Failed to install required base packages  ${BASE_PACKAGES[@]}"
+		fi
+		;;
+
+	8*)
+		log "INFO" "Setting specific part for v8 ($detected_os_version)"
+
+		RELEASE_RPM_URL="http://yum.centreon.com/standard/$CENTREON_MAJOR_VERSION/el8/stable/noarch/RPMS/centreon-release-$CENTREON_RELEASE_VERSION.el8.noarch.rpm"
+		PHP_BIN="/bin/php"
+		PHP_ETC="/etc/php.d"
+		OS_SPEC_SERVICES="php-fpm httpd"
+		PKG_MGR="dnf"
+
+		case "$detected_os_release" in
+
+		redhat-release*)
+			BASE_PACKAGES=(dnf-plugins-core epel-release)
+			subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
+			;;
+
+		centos-release* | centos-linux-release*)
+			BASE_PACKAGES=(dnf-plugins-core epel-release)
+			$PKG_MGR config-manager --set-enabled powertools
+			;;
+
+		oraclelinux-release* | enterprise-release*)
+			BASE_PACKAGES=(dnf-plugins-core oracle-epel-release-el8)
+			$PKG_MGR config-manager --set-enabled ol8_codeready_builder
+
+			;;
+		esac
+
+		#FIXME check the result
+		#FIXE handle the case where PHP 7.2 is already installed then switch to 7.3
+		log "INFO" "Installing PHP 7.3 and enable it"
+		$PKG_MGR module install php:7.3 -y -q
+		$PKG_MGR module enable php:7.3 -y -q
+
```

Comment:
php7.3 service failure should be managed

Reply:
we should better run with `set -e` in order to stop the script at any error
I thought it was the case
what do you think about it @quanghungb ?

Reply:
Considering:

- We are close to the release now
- This script has been widely tested so this issue does not seems to be happening at all
- This PR is a huge enhancement from the previous version of the unattended.sh

I think we should accept it like that and create a ticket for this enhancement.

Reply:
yes we could refactor the error management. I kept the original mechanism initially but the script was growing up since then. we will enhance it later as the change could be impacting now.